app/src/main/java/com/infomaniak/mail/ui/newMessage/AiViewModel.kt

```diff
@@ -89,6 +90,30 @@ class AiViewModel @Inject constructor(
         handleAiResult(apiResponse, userMessage, isUsingPreviousMessageAsContext = previousMessageBodyPlainText != null)
     }
 
+    fun splitBodyAndSubject(proposition: String): Pair<String?, String> {
+        return getSubjectAndContent(MATCH_SUBJECT_REGEX.find(proposition), proposition)
+    }
+
+    private fun getSubjectAndContent(match: MatchResult?, proposition: String): Pair<String?, String> {
+        val (subject, content) = match?.let {
+            // The method get on MatchGroupCollection is not available on API25
+            if (Build.VERSION.SDK_INT == Build.VERSION_CODES.N_MR1) {
+                val destructuredList = it.destructured.toList()
+                destructuredList[INDEX_AI_PROPOSITION_SUBJECT] to destructuredList[INDEX_AI_PROPOSITION_CONTENT].trim()
+            } else {
+                val subject = it.groups["subject"]?.value?.trim()
+                val content = it.groups["content"]?.value
+                subject to content
+            }
+        } ?: (null to proposition)
+
+        return if (subject.isNullOrBlank() || content == null) {
+            null to proposition
+        } else {
+            subject to content
+        }
+    }
```

Comment:
Can't we keep the method exactly as it was by just renaming it while annotating it with the API min level and make it so the new `splitBodyAndSubject` only does an if/else to choose between calling the actual method or doing nothing and returning the value directly?

Reply:
Can